### ISSUE TITLE
Events.getActor may return empty actor

### ISSUE THREAD

**Issue by yuri-sergiichuk** [SpineEventEngine/core-java#921]
Current functionality assumes that `ActorContext` is filled in the nearest to the event `CommandContext` and for all events emitted as a result of the command processing it is true. But, if an event was emitted as a result of the `@React`ion to another event, the nearest `CommandContext` of the current `EventContext` will be empty, because the `EventContext` `OriginCase` is equal to `EVENT_CONTEXT` is such case.

Maybe it worth changing implementation in a way, that will allow obtaining actor from the nested command contexts if needed, e.g. smth like that may work:
```
public static UserId getActor(EventContext context) {
        EventContext nestedContext = getNestedContext(context);
        CommandContext commandContext = nestedContext.getCommandContext();
        return commandContext.getActorContext()
                             .getActor();
    }

private static EventContext getNestedContext(EventContext context) {
        EventContext currentContext = context;
        while (currentContext.getOriginCase() == EventContext.OriginCase.EVENT_CONTEXT) {
            currentContext = currentContext.getEventContext();
        }
        return currentContext;
    }
```

**Comment by armiol** [comment]
We have to use `Events.findCommandContext()` in `Events.getActor()` implementation. And tweak `getActor` documentation accordingly — as currently it tells nothing about the `@React` use-case.

**Comment by armiol** [comment]
Also, we should check if there is a case where an empty actor is returned. I would return `Optional.empty()` if such a scenario is possible.

**Comment by dmitrykuzmin** [comment]
Resolved in #922.
